### DISCUSSION TITLE
How to disable auto centering of context menus?

### DISCUSSION THREAD

**Discussion by Bendrien** [hyprwm/Hyprland#6896] [Q&A] [answered]
This window type keeps getting centered on my machine which makes them unusable :(

For some you immediately loose focus because of that which makes them disappear again. For others with nested menus they just stack on top of each other.

How can I disable that?

```
Window 55b4659f4860 -> menu:
	mapped: 1
	hidden: 0
	at: 1729,1000
	size: 382,160
	workspace: 2 (2)
	floating: 1
	pseudo: 0
	monitor: 0
	class: REAPER
	title: menu
	initialClass: REAPER
	initialTitle: menu
	pid: 11151
	xwayland: 1
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	tags:
	swallowing: 0
	focusHistoryID: 1
```

**Comment by vaxerski** [comment]
xwayland moment. Is this a regression?

**Reply by Bendrien** [reply]
So far I had this only happening with REAPER, or what do you mean by that?

**Reply by Nicolas-Saavedra** [reply]
Yes, this is a regression. Doing a bisect right now, but REAPER works properly on 0.39, and fails to render menus properly on 0.41. Will notify later after bisect is done to have a more specific commit

**Reply by Nicolas-Saavedra** [reply]
Bisect shows this issue started happening on addd3e7f1. Seems fun. Should I make an issue or?

**Reply by vaxerski** [reply]
ye, please

**Reply by AmbrosYTC** [reply]
> So far I had this only happening with REAPER, or what do you mean by that?

Same issue, here's my temporary workaround in hyprland.conf:
```

windowrulev2 = float, class:REAPER

windowrulev2 = move cursor 5px 15px, class:REAPER

windowrulev2 = minsize 1 1, class:REAPER

windowrulev2 = stayfocused, class:REAPER
```

Not a super elegant solution but it gets the job done until a proper fix is deployed.

**Comment by Bendrien** [comment, marked as answer]
Issue has been fixed in the meanwhile!
